src/m365/spo/commands/site/site-recyclebinitem-list.ts

```diff
@@ -0,0 +1,101 @@
+import GlobalOptions from '../../../../GlobalOptions';
+import request from '../../../../request';
+import SpoCommand from '../../../base/SpoCommand';
+import commands from '../../commands';
+
+interface CommandArgs {
+  options: Options;
+}
+
+interface Options extends GlobalOptions {
+  siteUrl: string;
+  type?: string;
+  secondary?: boolean;
+}
+
+class SpoSiteRecycleBinItemListCommand extends SpoCommand {
+  public get name(): string {
+    return commands.SITE_RECYCLEBINITEM_LIST;
+  }
+
+  public get description(): string {
+    return 'Lists items from recycle bin';
+  }
+
+  public defaultProperties(): string[] | undefined {
+    return ['Id', 'Title', 'DirName'];
+  }
+
+  public commandAction(logger: Logger, args: CommandArgs, cb: (err?: any) => void): void {
+    if (this.verbose) {
+      logger.logToStderr(`Retrieving all items from recycle bin at ${args.options.siteUrl}...`);
+    }
+
+    const state: string = args.options.secondary ? '2' : '1';
+
+    let requestUrl: string = `${args.options.siteUrl}/_api/site/RecycleBin?$filter=(ItemState eq ${state})`;
+
+    if (typeof args.options.type !== 'undefined') {
+      const type = SpoSiteRecycleBinItemListCommand.recycleBinItemType.find(item => item.value === args.options.type);
+      if (typeof type !== 'undefined') {
+        requestUrl += ` and (ItemType eq ${type.id})`;
+      }
+    }
+
+    const requestOptions: any = {
+      url: requestUrl,
+      headers: {
+        'accept': 'application/json;odata=nometadata'
+      },
+      responseType: 'json'
+    };
+
+    request
+      .get<any>(requestOptions)
+      .then((value: any[] ): void => {
+        logger.log(value);
+        cb();
+      }, (err: any): void => this.handleRejectedODataJsonPromise(err, logger, cb));
+  }
+
+  public options(): CommandOption[] {
+    const options: CommandOption[] = [
+      {
+        option: '-u, --siteUrl <siteUrl>'
+      },
+      {
+        option: '-t, --type [type]',
+        autocomplete: SpoSiteRecycleBinItemListCommand.recycleBinItemType.map(item => item.value)
+      },
+      {
+        option: '-s, --secondary'
+      }
+    ];
+
+    const parentOptions: CommandOption[] = super.options();
+    return options.concat(parentOptions);
+  }
+
+  public validate(args: CommandArgs): boolean | string {
+    const isValidSharePointUrl: boolean | string = SpoCommand.isValidSharePointUrl(args.options.siteUrl);
+    if (isValidSharePointUrl !== true) {
+      return isValidSharePointUrl;
+    }
+
+    if (typeof args.options.type !== 'undefined' && !SpoSiteRecycleBinItemListCommand.recycleBinItemType.some(item => item.value === args.options.type)) {
+      return `${args.options.type} is not a valid value. Allowed values are ${SpoSiteRecycleBinItemListCommand.recycleBinItemType.map(item => item.value).join(', ')}`;
+    }
+
+    return true;
+  }
+
+  private static recycleBinItemType: { id: number, value: string }[] = [
+    { id: 1, value: 'listItems' },
```

Comment:
Are these types correct? I deleted a file and if I specify `--type file` it's not coming back, but it does when I specify `--type listItems`. Is that intended?

Reply:
@waldekmastykarz I will give it a check but are you sure you typed it correctly? I mean the type should be 'file**s**' not 'file' 🤔

Reply:
I double checked that and yes, I totally mixed all the numbers 😮.
but U corrected this and now is working fine. Please check the screen with tests after latest changes 
![image](https://user-images.githubusercontent.com/58668583/149677613-5c2bea8e-9d33-41b1-9968-d2c2fa6d8645.png)